lib/reverse_sentence.rb

```diff
@@ -1,6 +1,44 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is num of elements in string
+# Space complexity: O(1) no new space is created in algorithm
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  # nil check 
+  return my_sentence if my_sentence == nil || my_sentence.length == 0
```

Comment:
Regarding your comment: are `nil` and my_sentence.length both nil class? 
Consider the following code block:

```
words_without_length = ""
no_words = nil
puts words_without_length == no_words
puts words_without_length.class
puts no_words.class
puts words_without_length.equal? no_words
```